griptape/artifacts/text_artifact.py

```diff
@@ -11,23 +11,36 @@
     from griptape.tokenizers import BaseTokenizer
 
 
+def value_to_str(value: Any) -> str:
+    if isinstance(value, dict):
+        return "\n".join(f"{key}: {val}" for key, val in value.items())
+    else:
+        return str(value)
+
+
 @define
 class TextArtifact(BaseArtifact):
-    value: str = field(converter=str, metadata={"serializable": True})
+    value: str = field(converter=value_to_str, metadata={"serializable": True})
```

Comment:
any reason for the custom method? cant we just do 
```
value: str = field(converter=str, metadata={"serializable": True})
```
?

Reply:
Note the special formatting for dictionaries:
```python
    if isinstance(value, dict):
        return "\n".join(f"{key}: {val}" for key, val in value.items())
```
This removes a lot of the dictionary noise when conveying the artifact as text.

Reply:
i see. i dont disagree that its probably better for the LLM, but do we want to commit to potentially doing this for a bunch of other types too? what about `set` or `list`?

Reply:
Good point. Maybe this should live on the loader that creates the Artifact then?

Reply:
yeah that probably makes sense. the loader can choose to do special formatting tricks if needed.

Reply:
although im thinking that it would probably be _best_ to do it when it gets converted to a `MessageContent`, thats probably out of scope and too complex for this right now.

Reply:
That conversion happens by calling `Artifact.to_text()`. Honestly I think `CsvRowArtifact` actually works with this new model of Artifacts, but there are still too many issues with `value: dict`.

We could make `value` a property that automatically calls `to_text()`, but that feels like a hack.

Reply:
you mean changing value on the base class to:
```
_value: Any = field()

@property
def value(self):
    return self._value
```
that actually seems like a great idea.

Reply:
that would give child artifacts way more flexibility on init and when `.value` is called

Reply:
I meant updating `CsvRowArtifact` specifically to something like:
```python
_value: dict = field()

@property
def value(self):
    return self.to_text()
```

Not sure I fully follow your suggestion.

Reply:
i am suggesting the same thing, because the base class needs to be updated with:
```
_value: Any = field()

@property
def value(self):
    return self._value
```
in order for your suggestion to be possible.

I tried basically this same thing for `JsonArtifact` and i couldn't do it because of the base class

Reply:
Updated to have this use-case specific logic on `CsvRowArtifact`